adam/perception/perception_graph.py

```diff
@@ -501,7 +505,44 @@ def to_dot_node(pattern_node: "NodePredicate") -> str:
         dot_graph.render(output_file)
 
 
-@attrs(frozen=True, slots=True, eq=False)
+class DebugRenderingObject:
+    """
+        Helper callable class for debugging purposes. We render graph every hundred calls to match. The outputs can be
```

Comment:
style: Perhaps say something like "An instance of this object can be provided as the whatever-it-is argument of `GraphMatching.match` to render the match search process at every 100 time steps"

Reply:
Docstring should note that we don't start rendering for a certain time period